src/include/rbd/librbd.hpp

```diff
@@ -417,6 +425,9 @@ class CEPH_RBD_API RBD
   int group_image_list(IoCtx& io_ctx, const char *group_name,
                        std::vector<group_image_info_t> *images,
                        size_t group_image_info_size);
+  int group_image_list2(IoCtx& group_ioctx, const char *group_name,
+                            std::vector<cls::rbd::GroupImageStatus> *images,
```

Comment:
we do not expose cls structures via API. You need to introduce `rbd_group_image_status_t` for this. And may be call the method `group_image_status_list`?

But do we really want something like this in API? I think the status is an internal thing just to indicate something bad happened when adding(removing) image to a group.

Reply:
delete the API: group_image_list2

[https://github.com/ceph/ceph/pull/40808/commits/c058e51e97d50578a0e7c46b0bd66dd4f2e5db96](url)